src/Stratis.Features.FederatedPeg/TargetChain/FederatedPegBroadcaster.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Stratis.Bitcoin.Connection;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Features.FederatedPeg.Interfaces;
+using Stratis.Features.FederatedPeg.NetworkHelpers;
+
+namespace Stratis.Features.FederatedPeg.TargetChain
+{
+    public class FederatedPegBroadcaster : IFederatedPegBroadcaster
+    {
+        private readonly IConnectionManager connectionManager;
+        private readonly IFederatedPegSettings federatedPegSettings;
+
+        public FederatedPegBroadcaster(
+            IConnectionManager connectionManager,
+            IFederatedPegSettings federatedPegSettings)
+        {
+            this.connectionManager = connectionManager;
+            this.federatedPegSettings = federatedPegSettings;
+        }
+
+        /// <inheritdoc />
+        public async Task BroadcastAsync(Payload payload)
+        {
+            // TODO: Optimize how federation peers are identified.
+
+            List<INetworkPeer> peers = this.connectionManager.ConnectedPeers.ToList();
+
+            var ipAddressComparer = new IPAddressComparer();
+
+            foreach (INetworkPeer peer in peers)
```

Comment:
Could possibly use `ForeachAsync()` here to parallelise the broadcast, but it's a minor thing.